src/main/java/org/jenkinsci/plugins/workflow/cps/CpsFlowExecution.java

```diff
@@ -551,6 +560,15 @@ private Env createInitialEnv() {
                 return Envs.empty( isSandbox() ? new SandboxInvoker() : new DefaultInvoker());
             }
         });
+        } catch (Exception ex) { // Catch Exception instead of IOException because some XStream exceptions inherit from RuntimeException.
+            if (Thread.interrupted()) {
+                // Clear interrupt status so filesystem operations related to build completion succeed.
+                IOException interruption = new InterruptedIOException(this + " was interrupted while starting");
+                interruption.addSuppressed(ex);
+                throw interruption;
+            }
+            throw ex;
+        }
```

Comment:
Maybe this should be done in `workflow-job` instead? Should we copy any of [the code](https://github.com/jenkinsci/jenkins/blob/84153f66d2e93df4d9f5c457871e24a8cbf1f016/core/src/main/java/hudson/model/Run.java#L1907-L1912) in `Run.execute` related to interruptions? I need to test with `Executor.interrupt(Result, CauseOfInterruption)` to see if the `CauseOfInterruption` is reported correctly with the current code.

Reply:
I looked into this today, and yeah I think this catch block should be moved to `workflow-job` (with various changes to correctly report the causes of interruption), but the earlier check for `Thread.interrupted` needs to stay for interruption during startup to work as expected on Java 11.

Reply:
Moved to https://github.com/jenkinsci/workflow-job-plugin/pull/167.